sql/src/main/java/io/crate/operation/join/BitSet.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.join;
+
+import org.apache.lucene.util.LongBitSet;
+
+import java.util.ArrayList;
+
+/**
+ * This BitSet is used to mark matched rows between left and right in {@link NestedLoopOperation}
+ *
+ * Each bit true if the rows in the respective position are matched and therefore we need
+ * a structure capable of holding <pre>long</pre> size of bits so java.util.BitSet cannot be used.
+ *
+ * We choose to use {@link LongBitSet} from Lucence and add another layer of segments on top in
+ * order to further optimize performance. The usage of segments prevents from calling System.arrayCopy()
+ * and instead when the current segments are full we reserve a new segment and add it to the segments
+ * list.
+ */
+class BitSet {
+
+    private static long SEGMENT_SIZE = 1048576;
+    private static int INITIAL_NO_OF_SEGMENTS = 10;
```

Comment:
Lucene docs says LongBitSet should be used over FixedBitSet if more than 2.1B bits are stored, but we limit each bitset to only 1048576 but then initiate 10 of them? 
Does that really make sense?